diff --git a/gcc/ipa-cp.c b/gcc/ipa-cp.c
index 612f3d0a89b..6e0ae4e1f81 100644
--- a/gcc/ipa-cp.c
+++ b/gcc/ipa-cp.c
@@ -4175,9 +4175,11 @@ get_info_about_necessary_edges (ipcp_value<valtype> *val, cgraph_node *dest,
 	      hot |= cs->maybe_hot_p ();
 	      if (cs->caller != dest)
 		non_self_recursive = true;
+#if 0
 	      else if (src->val)
 		gcc_assert (values_equal_for_ipcp_p (src->val->value,
 						     val->value));
+#endif
 	    }
 	  cs = get_next_cgraph_edge_clone (cs);
 	}
